Backend/NoCode.FlowerShop.Application/Common/Behaviors/ValidationExtensions.cs

```diff
@@ -22,4 +23,21 @@ public static class ValidationExtensions
                 context.AddFailure("Password must contain at least one digit");
         });
     }
+    
+    public static IRuleBuilderOptionsConditions<T, string> ValidUrl<T>(this IRuleBuilder<T, string> ruleBuilder)
+    {
+        return ruleBuilder.Custom((url, context) =>
+        {
+            const string pattern 
+                = @"^[(http(s)?):\/\/(www\.)?a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)$";
+
+            var regex = MyRegex();
+
+            if (!regex.IsMatch(url)) 
+                context.AddFailure("The URL is not valid.");
+            });
+    }
+
+    [GeneratedRegex("^[(http(s)?):\\/\\/(www\\.)?a-zA-Z0-9@:%._\\+~#=]{2,256}\\.[a-z]{2,6}\\b([-a-zA-Z0-9@:%_\\+.~#?&//=]*)$")]
+    private static partial Regex MyRegex();
```

Comment:
MyRegex isn't really a good name, maybe `UrlRegex` sounds better?

Reply:
Rider doesn't do good job on naming things:D I will address this in a next PR